test/rowSelection.test.tsx

```diff
@@ -161,6 +220,9 @@ test('extra keys are preserved when updating the selectedRows Set', async () =>
   await toggleSelection(1);
   expect(set).toStrictEqual(new Set([...initialSet, 2]));
 
+  await userEvent.click(headerCheckbox);
+  expect(set).toStrictEqual(initialSet);
```

Comment:
All the rows are deselected when the indeterminate header checkbox is clicked. I think this behavior makes sense. Wdyt?

Reply:
There are other options
1. We can keep track of number of selected rows and if all the selectable rows are selected then clicking on the header checkbox can unselect them otherwise we can select all selectable rows
2. We can also consider selectable rows to determine if the header checkbox is selected. So if all the selectable rows are selected then mark the header checkbox as selected instead of indeterminate